icicle/tests/test_hash_api.cpp

```diff
@@ -313,38 +316,210 @@ class HashSumBackend : public HashBackend
   }
 };
 
-TEST_F(HashApiTest, MerkleTree)
+void assert_valid_tree(
+  const MerkleTree& tree,
+  int nof_input_byes,
+  const std::byte* inputs,
+  const std::vector<Hash>& hashes,
+  const MerkleTreeConfig& config)
+{
+  int nof_outputs = nof_input_byes * hashes[0].output_size() / hashes[0].input_default_chunk_size();
+  std::byte* layer_in = new std::byte[nof_input_byes]; // Going layer by layer - having the input layer as the largest
```

Comment:
you should always avoid new/delete in modern c++.
Use
`auto layer_in = std::make_unique<std::byte[]>(nof_inout_bytes); `
instead and no need to release. It also handles the case of exceptions